app/javascript/app/components/data-explorer-content/data-explorer-content-selectors.js

```diff
@@ -220,32 +224,111 @@ export const getLink = createSelector(
   }
 );
 
-function getOptions(section, filter, filtersMeta) {
-  if (section !== SECTION_NAMES.pathways) return filtersMeta[filter];
-  if (
-    filter === FILTER_NAMES.categories ||
-    filter === FILTER_NAMES.subcategories
-  ) {
-    return filtersMeta.categories;
+function getPathwaysModelOptions(query, filtersMeta, filter) {
+  if (!query || !query.locations) return filtersMeta[filter];
+  const locationsSelected = filtersMeta.locations.filter(
+    location => location.id === query.locations
+  );
+  const modelSelected = filtersMeta.models.filter(
+    model => model.id === query.models
+  );
+  const locationHasModel = modelSelected.find(model =>
+    model.geographic_coverage.includes(locationsSelected[0].name)
+  );
+  if (modelSelected.length && !locationHasModel) return filtersMeta[filter];
+  return filtersMeta[filter].filter(model =>
+    model.geographic_coverage.includes(locationsSelected[0].name)
+  );
+}
+
+function getPathwaysScenarioOptions(query, filtersMeta, filter) {
+  if (!query || !query.models) return filtersMeta[filter];
+  const selectedModelId = query.models;
+  return filtersMeta[filter].filter(
+    scenario => scenario.model.id === selectedModelId
+  );
+}
+
+function getPathwaysCategoryOptions(query, filtersMeta) {
+  if (!query || !query.scenarios) {
+    return filtersMeta.categories.filter(c => c.parent_id === null);
+  }
+  const selectedScenarioId = query.scenarios;
+  const scenarioIndicatorsIds = filtersMeta.scenarios.find(
+    sc => sc.id === selectedScenarioId
+  ).indicator_ids;
+  const categories = [];
+  scenarioIndicatorsIds.forEach(indId =>
```

Comment:
I thi

Reply:
⚡️ done 
Thanks, much cleaner that way (also spotted a bug 🐛 ⚔️ )